src/Stratis.BitcoinD/Program.cs

```diff
@@ -18,11 +18,20 @@ namespace Stratis.BitcoinD
 {
     public class Program
     {
+        /// <summary>The default port used by the API when the node runs on the bitcoin network.</summary>
+        private const int DefaultBitcoinApiPort = 37220;
```

Comment:
Why not just add ApiPort to the Network class (like RPC and solve all this work around much easier?)

Reply:
ApiPort on the network as a default for now would work great I think

Reply:
If we do that we may as well add all our network-specific feature settings to the network class. Unfortunately a user adding a new feature can't do that so it's important that we establish a mechanism that proves that this can be done and acts as an example to the user. We will just be postponing the inevitable and we will have a PR like this sooner or later.

Reply:
Sweet, nice one - I will follow your lead with CounterChainApiPort for sidechains too

Reply:
https://github.com/stratisproject/StratisBitcoinFullNode/issues/3465

Reply:
I am not sure about what you  mean with the `set the default` terminology, isn't what in network the default and of the user wants to overwrite it then they just specify this in the settings?

Reply:
`settings = command_line > config_file > defaultArgs > settings_constants_or_values_from_Network`.

From the above, if you set the `settings` directly then you remove the opportunity for the command line or config file to override the settings. Ideally you want to inject the network or application specific defaults where I have `defaultArgs`. If you happen to be lucky and there is a value on Network that the settings class can use then nothing needs to happen at `defaultArgs`. However that will not always be the case as we can't keep extending the network class for each setting. Features may be third party and those users can't (should not) update our Network class.